### PR TITLE
Update docs + database params, skip hmmsearch

### PR DESCRIPTION
**Pull request by jasmezz**
Update docs and database parameters, skip hmmsearch by default (to not have pipeline crash if user doesn't provide HMM files).

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/funcscan/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/funcscan _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [x] Usage Documentation in `docs/usage.md` is updated.
- [x] Output Documentation in `docs/output.md` is updated.
- [x] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: nextflow_schema.json [hunk 1]
```diff
@@ -382,275 +381,270 @@
...
             },
-            "fa_icon": "fas fa-tools",
-            "help_text": "Prodigal is a protein-coding gene prediction tool developed to run on bacterial and archaeal genomes.\n\nDocumentation: https://github.com/hyattpd/prodigal/wiki"
+            "fa_icon": "fas fa-file-signature"
         },
         "annotation_pyrodigal": {
             "title": "Annotation: Pyrodigal",
             "type": "object",
-            "description": "These parameters influence the annotation algorithm used by Pyrodigal.",
+            "description": "Pyrodigal produces protein-coding gene predictions of bacterial and archaeal genomes, based on the tool Prodigal being resource-optimized. Read more at the [Pyrodigal GitHub](https://github.com/althonos/pyrodigal) or its [documentation](https://pyrodigal.readthedocs.io).",
```
Comment (jfy133) [inline nextflow_schema.json:481]:
```suggestion
            "description": "Pyrodigal is a resource-optimized wrapper around Prodigal, producing protein-coding gene predictions of bacterial and archaeal genomes. Read more at the [Pyrodigal GitHub](https://github.com/althonos/pyrodigal) or its [documentation](https://pyrodigal.readthedocs.io).",
```

File: nextflow_schema.json [hunk 2]
```diff
@@ -683,45 +677,45 @@
...
+                    "help_text": "Assigns the file extension of the input files to allow AMPcombi2 to identify the tool output from the list of input files.\n\n> Modifies tool parameter(s):\n> - AMPCOMBI: `--hmmsearch_file`",
                     "fa_icon": "fas fa-address-card"
                 }
             },
-            "fa_icon": "fas fa-search-dollar"
+            "fa_icon": "fas fa-search"
         },
         "amp_ampcombi2_cluster": {
-            "title": "AMP: ampcombi2 cluster",
+            "title": "AMP: ampcombi2 cluster. Clusters the AMP candidates identified with AMPcombi. More info: [AMPcombi documentation](https://github.com/Darcy220606/AMPcombi) ",
```
Comment (jfy133) [inline nextflow_schema.json:716]:
```suggestion
            "title": "AMP: ampcombi2 cluster. Clusters AMP candidates identified with AMPcombi. More info: [AMPcombi documentation](https://github.com/Darcy220606/AMPcombi) ",
```

**Review by jfy133 — COMMENTED** [review]
First pass, will now load in launch and see how it looks

File: nextflow_schema.json [hunk 3]
```diff
@@ -683,45 +677,45 @@
...
             "type": "object",
-            "description": "Clustering submodule of AMPCOMBI2",
+            "description": "Clustering submodule of AMPcombi2",
```
Comment (jfy133) [inline nextflow_schema.json:718]:
```suggestion
            "description": "Clusters the AMP candidates identified with AMPcombi. More info: [AMPcombi documentation](https://github.com/Darcy220606/AMPcombi)",
```

File: nextflow_schema.json [hunk 4]
```diff
@@ -206,14 +206,14 @@
         "annotation_bakta": {
             "title": "Annotation: BAKTA",
             "type": "object",
-            "description": "These parameters influence the annotation algorithm of Bacteria used by BAKTA.",
```
Comment (jfy133) [inline nextflow_schema.json:209]: For the annotation section above the BAKTA specific ones, the tutle should be `Annotation: General options` (like with BGC)

File: nextflow_schema.json [hunk 5]
```diff
@@ -1076,43 +1062,63 @@
...
+                    "help_text": "Specifies the minimum coverage of the nucleotide sequence to be assigned an ARG hit using BLAST alignment. In the ABRicate matrix, an absent gene is assigned (`.`) and if present, it is assigned the estimated coverage (`#`).\n\nFor more information check the ABRicate [documentation](https://github.com/tseemann/abricate).\n\n> Modifies tool parameter(s):\n> - ABRicate: `--mincov`",
                     "fa_icon": "far fa-arrow-alt-circle-down",
                     "minimum": 1,
                     "maximum": 100
                 }
-            },
-            "help_text": "ABRicate screens for antimicrobial resistance or virulence genes based on NCBI, CARD, ARG-ANNOT, Resfinder, MEGARES, EcOH, PlasmidFinder, Ecoli_VF and VFDB databases.\n\nDocumentation: https://github.com/tseemann/abricate"
+            }
+        },
+        "bgc_general_filters": {
```
Comment (jfy133) [inline nextflow_schema.json:1072]:
```suggestion
        "bgc_general_options": {
```

Comment (jfy133) [inline nextflow_schema.json:1072]:
```suggestion
        "bgc_general_options": {
```

File: nextflow_schema.json [hunk 6]
```diff
@@ -1617,6 +1624,9 @@
         {
             "$ref": "#/definitions/arg_abricate"
         },
+        {
+            "$ref": "#/definitions/bgc_general_filters"
```
Comment (jfy133) [inline nextflow_schema.json:1628]:
```suggestion
            "$ref": "#/definitions/bgc_general_options"
```

File: nextflow_schema.json [hunk 7]
```diff
@@ -96,26 +96,26 @@
             "description": "These parameters influence the database to be used in classifying the taxonomy.",
             "default": "",
             "properties": {
-                "taxa_classification_mmseqs_databases_localpath": {
+                "taxa_classification_mmseqs_db": {
```
Comment (jfy133) [inline nextflow_schema.json:99]: I would set the `Taxonomic classifcication` general thing to `Taxonomic classificaton: general options` like the BGC one (see below, if you change the variable/entry name, make sure to update that everywhere)

File: nextflow_schema.json [hunk 8]
```diff
@@ -96,26 +96,26 @@
...
                     "description": "Specify a path to MMseqs2-formatted database.",
                     "help_text": "Specify a path to a database that is prepared in MMseqs2 format as detailed in the [documentation](https://mmseqs.com/latest/userguide.pdf).",
-                    "fa_icon": "fab fa-stackpath"
+                    "fa_icon": "fas fa-database"
```
Comment (jfy133) [inline nextflow_schema.json:102]: Line above is missing the `modifies tool parameter(s)` bit

File: nextflow_schema.json [hunk 9]
```diff
@@ -125,58 +125,58 @@
...
+                    "help_text": "This flag specifies whether the taxonomic lineage should be included in the output `.tsv` file. The taxonomic lineage is obtained from the internal module of `mmseqs/taxonomy` that infers the last common ancestor to classify the taxonomy. A value of '0' writes no taxonomic lineage, a value of '1' adds a column with the full lineage names prefixed with abbreviation of the lineage level, e.g. k_Prokaryotes;p_Bacteroidetes;c_....;o_....;f_....;g_....;s_...., while a value of '2' adds a column with the full NCBI taxids lineage,e.g. 1324;2345;4546;5345. More details can be found in the [documentation](https://mmseqs.com/latest/userguide.pdf).\n\n>  Modifies tool parameter(s):\n> - mmseqs taxonomy: `--tax-lineage`",
```
Comment (jfy133) [inline nextflow_schema.json:149]:
```suggestion
                    "help_text": "This flag specifies whether the taxonomic lineage should be included in the output `.tsv` file. The taxonomic lineage is obtained from the internal module of `mmseqs/taxonomy` that infers the last common ancestor to classify the taxonomy. A value of '0' writes no taxonomic lineage, a value of '1' adds a column with the full lineage names prefixed with abbreviation of the lineage level, e.g. `k_Prokaryotes;p_Bacteroidetes;c_....;o_....;f_....;g_....;s_....,` while a value of '2' adds a column with the full NCBI taxids lineage,e.g. `1324;2345;4546;5345`. More details can be found in the [documentation](https://mmseqs.com/latest/userguide.pdf).\n\n>  Modifies tool parameter(s):\n> - mmseqs taxonomy: `--tax-lineage`",
```

File: nextflow_schema.json [hunk 10]
```diff
@@ -206,14 +206,14 @@
...
+            "description": "BAKTA is a tool developed to annotate bacterial genomes and plasmids from both isolates and MAGs. More info: [Bakta documentation](https://github.com/oschwengers/bakta).",
             "default": "",
             "properties": {
-                "annotation_bakta_db_localpath": {
+                "annotation_bakta_db": {
                     "type": "string",
                     "fa_icon": "fas fa-database",
-                    "description": "Specify a path to BAKTA database.",
-                    "help_text": "Specify a path to a database that is prepared in a BAKTA format."
+                    "description": "Specify a path to a BAKTA database.",
```
[the remark this answers is not in the document]
Comment (jfy133) [reply nextflow_schema.json:215]: Missing 'modifies tool parameters' (check for everything)

File: nextflow_schema.json (hunk 8 above)
Comment (jfy133) [reply nextflow_schema.json:102]: Also check throughout everywhere if we are doing the tool naming by the process name or the tool itself (e.g. `MMSEQS_TAXONOMY` vs `mmseqs taxonomy`. The latter makes more sense from  a user PoV if tey go back to check the docs of the tool itself

Comment (jfy133) [reply nextflow_schema.json:102]: (or just the 'official' tool name... maybe that's what we went with...)

File: nextflow_schema.json [hunk 11]
```diff
@@ -259,20 +259,20 @@
                 "annotation_bakta_complete": {
```
Comment (jfy133) [inline nextflow_schema.json:259]: `annotation_bakta_singlemode` above has a badly formatting 'modifies too parameers' thing, it's missing a new line

File: nextflow_schema.json [hunk 12]
```diff
@@ -382,275 +381,270 @@
...
                     "description": "Path to AMPcombi reference database directory (DRAMP).",
-                    "help_text": "AMPcombi uses the 'general AMPs' dataset of the (DRAMP database)[http://dramp.cpu-bioinfor.org/downloads/] for taxonomic classification. If you have a local version of it, you can provide the path to the folder containing the reference database files:\n1. a fasta file with a `.fasta` file extension\n2. the corresponding table with with functional and taxonomic classifications in `.tsv` file extension.\n\nFor more information check AMPcombi [documentation](https://github.com/Darcy220606/AMPcombi).",
+                    "help_text": "AMPcombi uses the 'general AMPs' dataset of the [DRAMP database](http://dramp.cpu-bioinfor.org/downloads/) for taxonomic classification. If you have a local version of it, you can provide the path to the folder containing the reference database files:\n1. fasta file with `.fasta` file extension\n2. the corresponding table with with functional and taxonomic classifications in `.tsv` file extension.\n\nFor more information check the AMPcombi [documentation](https://github.com/Darcy220606/AMPcombi).",
```
Comment (jfy133) [inline nextflow_schema.json:633]:
```suggestion
                    "help_text": "AMPcombi uses the 'general AMPs' dataset of the [DRAMP database](http://dramp.cpu-bioinfor.org/downloads/) for taxonomic classification. If you have a local version of it, you can provide the path to the directory(!) that contains the following reference database files:\n1. fasta file with `.fasta` file extension\n2. the corresponding table with with functional and taxonomic classifications in `.tsv` file extension.\n\nFor more information check the AMPcombi [documentation](https://github.com/Darcy220606/AMPcombi).",
```

File: nextflow_schema.json [hunk 13]
```diff
@@ -768,43 +762,43 @@
...
                 "arg_skip_amrfinderplus": {
                     "type": "boolean",
-                    "description": "Skip AMRFinderPlus during the ARG-screening.",
+                    "description": "Skip AMRFinderPlus during the ARG screening.",
                     "fa_icon": "fas fa-ban"
                 },
                 "arg_amrfinderplus_db": {
                     "type": "string",
                     "fa_icon": "fas fa-layer-group",
                     "help_text": "Specify the path to a local version of the ARMFinderPlus database. If no input is given, the pipeline will download the database for you.\n\n See the nf-core/funcscan usage [documentation](https://nf-co.re/funcscan/usage) for more information.",
```
Comment (jfy133) [inline nextflow_schema.json:787]: Missing 'modifies tool' section

File: nextflow_schema.json [hunk 14]
```diff
@@ -835,91 +829,89 @@
...
                     "fa_icon": "fas fa-ban"
                 },
-                "arg_deeparg_data": {
+                "arg_deeparg_db": {
                     "type": "string",
-                    "fa_icon": "fab fa-deezer",
+                    "fa_icon": "fas fa-database",
                     "description": "Specify the path to the DeepARG database.",
-                    "help_text": "Specify the path to a local version of the DeepARG database (see the pipelines' usage [documentation](https://nf-co.re/funcscan/usage)). If no input is given, the module will download the database for you, however this is not recommended, as the database is large and this will take time."
+                    "help_text": "Specify the path to a local version of the DeepARG database (see the pipelines' usage [documentation](https://nf-co.re/funcscan/dev/docs/usage#databases-and-reference-files)). If no input is given, the module will download the database for you, however this is not recommended, as the database is large and this will take time."
```
Comment (jfy133) [inline nextflow_schema.json:844]: Missing 'modifies tool paramers' section

File: nextflow_schema.json [hunk 15]
```diff
@@ -347,33 +347,32 @@
...
             },
-            "fa_icon": "fas fa-file-signature",
-            "help_text": "BAKTA is a tool developed to annotate bacterial genomes and plasmids from both isolates and MAGs. \n\nDocumentation: [https://github.com/oschwengers/bakta](https://github.com/oschwengers/bakta)"
+            "fa_icon": "fas fa-file-signature"
         },
         "annotation_prokka": {
             "title": "Annotation: Prokka",
             "type": "object",
-            "description": "These parameters influence the annotation algorithm used by Prokka.",
+            "description": "Prokka annotates genomic sequences belonging to bacterial, archaeal and viral genomes. More info: [Prokka documentation](https://github.com/tseemann/prokka)",
```
Comment (jfy133) [inline nextflow_schema.json:355]: Markdown links don't work in the descriptions, replace with a raw link

Comment (jfy133) [reply nextflow_schema.json:355]: image

Applies to all sections!

File: nextflow_schema.json [hunk 16]
```diff
@@ -1142,43 +1148,42 @@
...
-                    "description": "Skip deepBGC during the BGC screening."
+                    "description": "Skip DeepBGC during the BGC screening."
                 },
-                "bgc_deepbgc_database": {
+                "bgc_deepbgc_db": {
                     "type": "string",
-                    "fa_icon": "fas fa-layer-group",
-                    "description": "Path to local deepBGC database folder."
+                    "fa_icon": "fas fa-database",
+                    "description": "Path to local DeepBGC database folder."
```
Comment (jfy133) [inline nextflow_schema.json:1186]: Missing `modifies path`

**Review by jfy133 — CHANGES_REQUESTED** [review]
You have very inconsistent icons, e.g. AMP have screwdriver/wrench icon, except AMPCombi which have two different ones, ARG are all the microbe icon (find), BGC has a mixture of microbes and screwdriver/wrench icon (antiSMASH). Classification is also mixed, whereas annotation is all the same.

I would suggest having one icon per ARG/BGC/AMP category.

File: docs/usage.md [hunk 17]
```diff
@@ -159,10 +159,10 @@ nf-core/funcscan allows screening of sequences for functional genes associated w
 
 This requires supplying a list of HMM files ending in `.hmm`, that have models for the particular molecule(s) or BGCs you are interested in. You can download these files from places such as [PFAM](https://www.ebi.ac.uk/interpro/download/Pfam/) for antimicrobial peptides (AMP), or the antiSMASH GitHub repository for [biosynthetic gene cluster](https://github.com/antismash/antismash/tree/master/antismash/detection/hmm_detection/data) related HMMs, or create them yourself.
 
-You should place all HMMs in a directory and supply them e.g. to AMP models:
+You should place all HMMs in a directory, supply them to the AMP or BGC workflow and switch hmmsearch on like:
 
 ```bash
---amp_hmmsearch_models '/<path>/<to>/<amp>/*.hmm'
+--amp_run_hmmsearch --amp_hmmsearch_models '/<path>/<to>/<amp>/*.hmm'
```
Comment (jasmezz) [inline docs/usage.md:165]:
```suggestion
--amp_run_hmmsearch --amp_hmmsearch_models "/<path>/<to>/<amp>/*.hmm"
```

File: CHANGELOG.md [hunk 18]
```diff
@@ -28,6 +28,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - [#371](https://github.com/nf-core/funcscan/pull/371) Fixed AMRFinderPlus parameter `arg_amrfinderplus_name`. (by @m3hdad)
 - [#376](https://github.com/nf-core/funcscan/pull/376) Fixed an occasional RGI process failure when certain files not produced. (❤️ to @amizeranschi for reporting, fix by @amizeranschi & @jfy133)
 - [#386](https://github.com/nf-core/funcscan/pull/386) Updated DeepBGC module to fix output file names, separate annotation step for all BGC tools, add warning if no BGCs found, fix MultiQC reporting of annotation workflow. (by @jfy133, @jasmezz)
+- [#391](https://github.com/nf-core/funcscan/pull/391) Skip hmmmsearch by default to not crash pipeline if user provides no HMM files, updated all "database" parameters, updated docs. (by @jasmezz)
```
Comment (jasmezz) [inline CHANGELOG.md:31]:
```suggestion
- [#391](https://github.com/nf-core/funcscan/pull/391) Skip hmmmsearch by default to not crash pipeline if user provides no HMM files, updated docs. (by @jasmezz)
- [#391](https://github.com/nf-core/funcscan/pull/391) Unified all "database" parameters. (by @jasmezz)
```

File: nextflow_schema.json (hunk 8 above)
Comment (jasmezz) [reply nextflow_schema.json:102]: Yep, it's the tool name in all cases :)

Comment (jasmezz) [reply nextflow_schema.json:102]: I checked again, this doesn't modify a tool parameter. Rather in the pipeline logic decided whether a DB will be downloaded or the DB from this param being used.

File: nextflow_schema.json (hunk 11 above)
Comment (jasmezz) [reply nextflow_schema.json:259]: That's weird. The code looks exactly like that of the other params. Do you have a screenshot?

File: nextflow_schema.json (hunk 8 above)
Comment (jasmezz) [reply nextflow_schema.json:102]: Okay added the correct environment variable anyway ;)

File: nextflow_schema.json (hunk 11 above)
Comment (jasmezz) [reply nextflow_schema.json:259]: In my `nf-core launch` it looks alright:
image

File: CHANGELOG.md [hunk 19]
```diff
@@ -28,7 +28,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - [#371](https://github.com/nf-core/funcscan/pull/371) Fixed AMRFinderPlus parameter `arg_amrfinderplus_name`. (by @m3hdad)
 - [#376](https://github.com/nf-core/funcscan/pull/376) Fixed an occasional RGI process failure when certain files not produced. (❤️ to @amizeranschi for reporting, fix by @amizeranschi & @jfy133)
 - [#386](https://github.com/nf-core/funcscan/pull/386) Updated DeepBGC module to fix output file names, separate annotation step for all BGC tools, add warning if no BGCs found, fix MultiQC reporting of annotation workflow. (by @jfy133, @jasmezz)
-- [#391](https://github.com/nf-core/funcscan/pull/391) Skip hmmmsearch by default to not crash pipeline if user provides no HMM files, updated all "database" parameters, updated docs. (by @jasmezz)
+- [#391](https://github.com/nf-core/funcscan/pull/391) Skip hmmmsearch by default to not crash pipeline if user provides no HMM files, updated docs. (by @jasmezz)
+- [#391](https://github.com/nf-core/funcscan/pull/391) Unified all "database" parameters. (by @jasmezz)
```
Comment (jfy133) [inline CHANGELOG.md:34]:
```suggestion
- [#391](https://github.com/nf-core/funcscan/pull/391) Made all "database" parameter names consistent. (by @jasmezz)
```

File: docs/usage.md [hunk 20]
```diff
@@ -162,7 +162,7 @@ This requires supplying a list of HMM files ending in `.hmm`, that have models f
 You should place all HMMs in a directory, supply them to the AMP or BGC workflow and switch hmmsearch on like:
 
 ```bash
---amp_run_hmmsearch --amp_hmmsearch_models '/<path>/<to>/<amp>/*.hmm'
+--amp_run_hmmsearch --amp_hmmsearch_models "/<path>/<to>/<amp>/*.hmm"
 ```
```
Comment (jfy133) [inline docs/usage.md:167]:
```suggestion

:::warning
Ensure to wrap in quotes if using an asterisk, to ensure Nextflow (not your shell) parses the wildcard
:::
```

**Review by jfy133 — APPROVED** [review]
I made a few more changes, but otherwise you can now merge @jasmezz , we will find anything else later but they will be minor :grimacing:

**Pull request merged**